### DISCUSSION TITLE
FontFamily font metrics compatible functions

### DISCUSSION THREAD

**Discussion by jahav** [SixLabors/Fonts#299] [Ideas] [answered]
Since System.Drawing.Common is dead (=[runtime exceptions galore](https://learn.microsoft.com/en-us/dotnet/core/compatibility/core-libraries/6.0/system-drawing-common-windows-only)), it would be benefical to provide methods compatible with FontFamily metrics for easier migration from System.Drawing.Common to SixLabors.Fonts. 

`FontFamily.GetCellAscent` and `FontFamily.GetCellDescent` return values from OS/2 metrics WinAscent and WinDescent (e.g. System.Drawing.Common for Calibri font returns descent 550 and ascent 1950, while all SixLabors.Fonts (+FontForge and others) return (probably more correct) ascent 1536 and descent 512). About 30% of system fonts return different metric values.

I looked through the code and it seems that the reflection of OS2Table is currently the only way.

Use case: In my case, I need to be compatible with the Excel adjust-cell-size-to-content that uses FontFamily values. I will take reflection, but compatible API would be nice.

**Comment by JimBobSquarePants** [comment, marked as answer]
These properties already exist in the `FontMetrics` type which is a property of the `Font` type. https://docs.sixlabors.com/api/Fonts/SixLabors.Fonts.FontMetrics.html

You want the normalised values as there are properties within a font file which determine which metric to use. In addition some fonts miss certain metrics which require replacement values to be calculated based on known conventions. 

There no benefit to us exposing the internal metrics. 

TL;DR System.Drawing is wrong. Six Labors and FontForge are correct.
